AsyncDisplayKit/Details/_ASDisplayViewAccessiblity.mm

```diff
@@ -0,0 +1,153 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "_ASDisplayViewAccessiblity.h"
+#import "_ASDisplayView.h"
+#import "ASDisplayNode+FrameworkPrivate.h"
+
+#import <objc/runtime.h>
+#import <queue>
+
+
+#pragma mark - UIAccessibilityElement
+
+static const char *ASDisplayNodeAssociatedNodeKey = "ASAssociatedNode";
+
+@implementation UIAccessibilityElement (_ASDisplayView)
+
+- (void)setAsyncdisplaykit_node:(ASDisplayNode *)node
+{
+  objc_setAssociatedObject(self, ASDisplayNodeAssociatedNodeKey, node, OBJC_ASSOCIATION_ASSIGN); // Weak reference to avoid cycle, since the node retains the layer.
+  
+  // Update UIAccessibilityElement properties from node
+  self.accessibilityIdentifier = node.accessibilityIdentifier;
+  self.accessibilityLabel = node.accessibilityLabel;
+  self.accessibilityHint = node.accessibilityHint;
+  self.accessibilityValue = node.accessibilityValue;
+  self.accessibilityTraits = node.accessibilityTraits;
+}
+
+- (ASDisplayNode *)asyncdisplaykit_node
+{
+  return objc_getAssociatedObject(self, ASDisplayNodeAssociatedNodeKey);
+}
+
+@end
+
+
+#pragma mark - _ASDisplayView
+
+@interface _ASDisplayView () {
+  NSMutableArray *_accessibleElements;
+}
+
+@end
+
+@implementation _ASDisplayView (UIAccessibilityContainer)
+
+#pragma mark - UIAccessibility
+
+- (NSArray *)accessibleElements
+{
+  ASDisplayNode *selfNode = self.asyncdisplaykit_node;
+  if (selfNode == nil) {
+    return nil;
+  }
+  
+  _accessibleElements = [[NSMutableArray alloc] init];
+  
+  // Handle rasterize case
+  if (selfNode.shouldRasterizeDescendants) {
+    // In this case we have to go through the whole subnodes tree in BFS fashion and create all
+    // accessibility elements ourselves as the view hierarchy is flattened
+    
+    // Queue used to keep track of subnodes while traversing this layout in a BFS fashion.
+    std::queue<ASDisplayNode *> queue;
+    queue.push(selfNode);
+    
+    while (!queue.empty()) {
+      ASDisplayNode *node = queue.front();
+      queue.pop();
+      
+      // Check if we have to add the node to the accessiblity nodes as it's an accessiblity element
```

Comment:
This is well done. Perhaps we should provide a convenient method that allows running a block on each node in BFS search fashion?